atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueCacheImpl.java

```diff
@@ -202,8 +210,17 @@ private void clearCache() {
     }
 
     private static Stream<CellReference> extractTableAndCell(LockDescriptor descriptor) {
-        // TODO(jshah): this has potentially large blow-up for cells with lots of zero bytes. We should probably make
-        //  users opt-in or warn when this is the case.
-        return AtlasLockDescriptorUtils.candidateCells(descriptor).stream();
+        List<CellReference> candidateCells = AtlasLockDescriptorUtils.candidateCells(descriptor);
```

Comment:
We can still pass in the actual schema and only try to match with the cached tables from the schema if this is something that we actually hit

Reply:
Let's leave that optimisation for now, but possibly something we can visit later.